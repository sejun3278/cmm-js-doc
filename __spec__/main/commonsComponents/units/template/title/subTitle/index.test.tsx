import { render } from "@testing-library/react";

import _SubTitleTemplate from "src/main/commonsComponents/units/template/title/subTitle";

export default describe("subTitle Template Page", () => {
  // 스냅샷
  test("subTitle Template Page - Snapshot", () => {
    const { container } = render(
      <_SubTitleTemplate title="서브타이틀 테스트">
        <div>내용</div>
      </_SubTitleTemplate>
    );
    expect(container).toMatchSnapshot();
  });

  // h2 태그 여부 확인 및 텍스트 검증
  test("subTitle Template Page - Check have h2 tag and title", () => {
    const { container } = render(
      <_SubTitleTemplate title="사용 방법">
        <div className="contents">내용</div>
      </_SubTitleTemplate>
    );

    // h2 태그를 감싸고 있는 wrapper의 존재여부 검증
    const wrapperEle = container.querySelector("._subTitleTemplate_");
    expect(wrapperEle).toBeInTheDocument();

    // Wrapper안에 h2태그 존재 여부 검증
    const h2Ele = wrapperEle?.querySelector(
      "._subTitle_"
    ) as HTMLHeadingElement;
    expect(h2Ele).toBeInTheDocument();

    if (h2Ele) {
      expect(h2Ele.innerText).toEqual("🔍 사용 방법"); // 타이틀 검증
    }
  });

  // wrapper 안에 img 태그 존재 여부 검증
  test("subTitle Template Page - Check have img tag in wrapper", () => {
    const { container } = render(
      <_SubTitleTemplate title="사용 방법">
        <img
          alt=""
          src="/images/commons/logo/MCM_main_logo.png"
          className="img"
        />
      </_SubTitleTemplate>
    );

    // h2 태그를 감싸고 있는 wrapper의 존재여부 검증
    const wrapperEle = container.querySelector("._subTitleTemplate_");
    expect(wrapperEle).toBeInTheDocument();

    if (wrapperEle) {
      // img 태그 존재 여부 검증
      const imgEle = wrapperEle.querySelector(".img");
      expect(imgEle).toBeInTheDocument();
      expect(imgEle).toHaveClass("img");
      expect(imgEle).toHaveAttribute(
        "src",
        "/images/commons/logo/MCM_main_logo.png"
      );
    }
  });
});
