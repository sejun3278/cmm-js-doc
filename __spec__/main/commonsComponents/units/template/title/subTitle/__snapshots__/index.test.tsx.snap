// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`subTitle Template Page subTitle Template Page - Snapshot 1`] = `
<div>
  <div
    class="_subTitleTemplate_ css-k59gj9"
  >
    <div>
      <h2
        class="_title_ _subTitle_"
      />
    </div>
    <div
      class="_subTitleLine_ css-18klhw4"
    />
    <div>
      내용
    </div>
  </div>
</div>
`;
